src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    Animal.alive.push(this);
+  }
+
+  bite(obj) {
+    if (!obj.hidden && !(obj instanceof Carnivore)) {
+      obj.health -= 50;
+      Animal.alive = Animal.alive.filter(animal => animal.health > 0);
```

Comment:
You are doing filtration always, for every bite
It will be better to do it only when it was deadly bite (when obj.health <= 0)